changes/951.fixed

```diff
@@ -0,0 +1 @@
+Check if the data isn't already in database and retry
```

Comment:
Please update the changelog fragment according to the [best practices](https://docs.nautobot.com/projects/core/en/stable/development/core/#creating-changelog-fragments)